tracer/src/Datadog.Trace/AppSec/Security.cs

```diff
@@ -136,12 +136,38 @@ public static Security Instance
 
         internal string? DdlibWafVersion => _waf?.Version;
 
-        internal bool TrackUserEvents => Enabled && Settings.UserEventsAutomatedTracking != "disabled";
+        internal bool IsTrackUserEventsEnabled =>
+            Enabled && CalculateIsTrackUserEventsEnabled(_configurationStatus.AutoUserInstrumMode, Settings.UserEventsAutoInstrumentationMode);
 
-        internal bool IsExtendedUserTrackingEnabled => Settings.UserEventsAutomatedTracking == SecuritySettings.UserTrackingExtendedMode;
+        internal bool IsAnonUserTrackingMode => CalculateIsAnonUserTrackingMode(_configurationStatus.AutoUserInstrumMode, Settings.UserEventsAutoInstrumentationMode);
 
         internal ApiSecurity ApiSecurity { get; }
 
+        internal static bool CalculateIsTrackUserEventsEnabled(string? remote, string local)
+        {
+            if (remote is SecuritySettings.UserTrackingIdentMode or SecuritySettings.UserTrackingAnonMode)
+            {
+                return true;
+            }
+
+            if (remote is SecuritySettings.UserTrackingDisabled or not null)
+            {
+                return false;
+            }
+
+            return local is SecuritySettings.UserTrackingIdentMode or SecuritySettings.UserTrackingAnonMode;
```

Comment:
If local is undefined and remote is also undefined, should we return true (identification mode enabled)?

Reply:
Local can never be null, we handle defaults in the setting class (so they will be recorded by telemetry). Added a comment. See 266b7a62b.